### ISSUE TITLE
Use ARRAY_JOB_ID instead of JOB_ID in mail subject

### ISSUE THREAD

**Issue by noebi** [neilmunday/slurm-mail#135]
Hi,
we got reports from confused users who did not really understand why
the mail subject shows a job id for their array job that they've never seen
before and one that's different from the one displayed in the mail body.

Not quite sure why, but in our local setup, task 0 gets a JOB_ID that is equal to 
ARRAY_JOB_ID + 1.

So we decided to patch slurm-mail locally to make the subject line use ARRAY_JOB_ID 
for array jobs.

Patch:
```
--- ./src/slurmmail/cli.py.orig 2024-06-12 22:51:04.000000000 +0200
+++ ./src/slurmmail/cli.py      2024-06-26 13:13:50.394081938 +0200
@@ -761,9 +761,14 @@
             subject_state = state
 
         msg = MIMEMultipart("alternative")
-        msg["Subject"] = Template(options.email_subject).substitute(
-            CLUSTER=job.cluster, JOB_ID=job.id, JOB_NAME=job.name, STATE=subject_state
-        )
+        if job.is_array():
+            msg["Subject"] = Template(options.email_subject).substitute(
+                CLUSTER=job.cluster, JOB_ID=job.array_id, JOB_NAME=job.name, STATE=subject_state
+            )
+        else:
+            msg["Subject"] = Template(options.email_subject).substitute(
+                CLUSTER=job.cluster, JOB_ID=job.id, JOB_NAME=job.name, STATE=subject_state
+            )
         msg["To"] = user_email
         msg["From"] = options.email_from_address
         msg["Date"] = email.utils.formatdate(localtime=True)

```

**Comment by neilmunday** [comment]
Issue fixed in 4.20 branch and will be included in the 4.20 release.

**Comment by neilmunday** [comment]
Version 4.20 released.
